### ISSUE TITLE
get_content() behaviour and a potential improvement

### ISSUE THREAD

**Issue by AitorF** [simplepie/simplepie#478]
Found a feed with an item structure like the following:

```
<description>
    <![CDATA[
        Here some description of the item
    ]]>
</description>
<content:encoded>
    <![CDATA[ ]]>
</content:encoded>
```

For this item, get_content() is returning the whitespace string inside `<content:encoded>` tag. I guess the publisher could do the things better, but should get_content() try to deal with this??

Below changes in `get_content()` solves the issue but not sure how far should SimplePie try to go to solve what seems to be a publisher mistake.

```
public function get_content($content_only = false)
{
    $atom10Content = $this->get_item_tags(SIMPLEPIE_NAMESPACE_ATOM_10, "content");
    $atom03Content = $this->get_item_tags(SIMPLEPIE_NAMESPACE_ATOM_03, "content");
    $rss10Content = $this->get_item_tags(SIMPLEPIE_NAMESPACE_RSS_10_MODULES_CONTENT, "encoded");
    $description = $this->get_description(true);
    if ($atom10Content && trim($atom10Content[0]["data"]))
    {
        return $this->sanitize($atom10Content[0]["data"], $this->registry->call("Misc", "atom_10_content_construct_type", array($atom10Content[0]["attribs"])), $this->get_base($atom10Content[0]));
    }
    elseif ($atom03Content && trim($atom03Content[0]["data"]))
    {
        return $this->sanitize($atom03Content[0]["data"], $this->registry->call("Misc", "atom_03_construct_type", array($atom03Content[0]["attribs"])), $this->get_base($atom03Content[0]));
    }
    elseif ($rss10Content && trim($rss10Content[0]["data"]))
    {
        return $this->sanitize($rss10Content[0]["data"], SIMPLEPIE_CONSTRUCT_HTML, $this->get_base($rss10Content[0]));
    }
    elseif (!$content_only)
    {
        return $description;
    }
    else
    {
        return null;
    }
}
```

**Comment by mblaney** [comment]
Hi @AitorF you shouldn't need to refactor it like that. Just work out why it isn't falling through to get_description() already and we can fix that problem.

**Comment by JanPetterMG** [comment]
Checking for empty content should be mandatory if you ask me!

Yesterday I've discovered large number of feeds that has fairly large numbers of whitespaces (spaces, tabs, newlines, etc) in both the start and end of the items encoded description. Is there a reason why the returned data isn't trimmed in the first place?

Haven't checked the source code yet, but if `<content:encoded>` is empty, as an tag with no characters in it at all, that wouldn't replace an non-empty `<description>` right?

**Comment by mblaney** [comment]
that's right, `<content:encoded>` would need to be empty for it to be replaced with `<description>`. So both issues mentioned here can be solved by calling `trim` before returning in `get_item_tags`.

**Comment by JanPetterMG** [comment]
What I don't understand is why `get_content()` returns content that's not trimmed, when it's calling `sanitize()` witch trims the data...
This complexity.... arrrgh

**Comment by AitorF** [comment]
@mblaney that's just what my code above tries to deal with. In my super-short research, I didn't find a better way to manage it.

**Comment by mblaney** [comment]
Fixed by #494
